corehq/ex-submodules/casexml/apps/stock/models.py

```diff
@@ -61,6 +61,14 @@ def __unicode__(self):
             case=self.case_id, product=self.product_id, section_id=self.section_id,
         )
 
+    def save(self, force_insert=False, force_update=False, using=None, update_fields=None):
+        # this is a bit hacky, but allows us to create StockReport and StockTransaction objects
+        # in a place that isn't exactly where we save them, which fits better with the current
+        # form processing workflows
+        if self.report_id is None and self.report and self.report.id is not None:
+            self.report_id = self.report.id
+        super(StockTransaction, self).save(force_insert=force_insert)
```

Comment:
the issue is when you create the report at the same time as the transaction django initializes `report_id` once at whatever it was, rather than doing it just in time. the way this works is that the report ends up getting saved between when this model is initialized but before it's saved (so it doesn't have an `id` when it's assigned but does by the time this `.save()` gets called). do you have a better way to deal with this?